x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -581,28 +589,37 @@ private LogicalPlan resolveEnrich(Enrich enrich, List<Attribute> childrenOutput)
         }
     }
 
-    private static List<Attribute> resolveAgainstList(UnresolvedAttribute u, Collection<Attribute> attrList) {
-        var matches = AnalyzerRules.maybeResolveAgainstList(u, attrList, false, true, Analyzer::handleSpecialFields);
+    private static List<Attribute> resolveAgainstList(UnresolvedNamePattern up, Collection<Attribute> attrList) {
+        UnresolvedAttribute ua = new UnresolvedAttribute(up.source(), up.pattern(), null);
+        Predicate<Attribute> matcher = a -> up.match(a.name()) || up.match(a.qualifiedName());
+        var matches = AnalyzerRules.maybeResolveAgainstList(matcher, () -> ua, attrList, true, a -> Analyzer.handleSpecialFields(ua, a));
+        return potentialCandidatesIfNoMatchesFound(ua, matches, attrList, list -> UnresolvedNamePattern.errorMessage(up.pattern(), list));
+    }
+
+    private static List<Attribute> resolveAgainstList(UnresolvedAttribute ua, Collection<Attribute> attrList) {
+        var matches = AnalyzerRules.maybeResolveAgainstList(ua, attrList, a -> Analyzer.handleSpecialFields(ua, a));
+        return potentialCandidatesIfNoMatchesFound(ua, matches, attrList, list -> UnresolvedAttribute.errorMessage(ua.name(), list));
+    }
 
+    private static List<Attribute> potentialCandidatesIfNoMatchesFound(
+        UnresolvedAttribute ua,
+        List<Attribute> matches,
+        Collection<Attribute> attrList,
+        java.util.function.Function<List<String>, String> messageProducer
+    ) {
         // none found - add error message
         if (matches.isEmpty()) {
-            UnresolvedAttribute unresolved;
-            var name = u.name();
-            if (Regex.isSimpleMatchPattern(name)) {
-                unresolved = u.withUnresolvedMessage(format(null, "No match found for [{}]", name));
-            } else {
-                Set<String> names = new HashSet<>(attrList.size());
-                for (var a : attrList) {
```

Comment:
Most of the changes here are massaging the previous code to work without an `UnresolvedAttribute`.
A bool parameter was not used and the extension was done through a supplier in order to throw an error when the pattern does not match.